enhancements/api-review/custom-tags-aws.md

```diff
@@ -73,71 +95,373 @@ Note existing unchanged behavior: The installer will apply these tags to all AWS
+
+- As a cluster administrator of OpenShift, I expect user-defined tags added in Infrastructure CR are reconciled and desired user-defined tags maintained on AWS resources.
+  The latest updates on `.spec.platformSpec.aws.resourceTags` field must be reconciled to AWS resources and desired user-defined tags maintained when being managed by OpenShift.
+  Any modifications which are external to OpenShift must be reconciled to desired user-defined tags from `.spec.platformSpec.aws.resourceTags`.
+
+The following user stories are added to support existing methods of updating user-defined tags which override `.spec.platformSpec.aws.resourceTags` to avoid breaking change.
+As the existing cluster might be configured with user-defined tags from kube resources, the precedence is given to user-defined tags in kube resources.
+- As a cluster administrator of OpenShift, I expect user-defined tags added in kube resources override the entries in `.spec.platformSpec.aws.resourceTags`.
+
+- As a cluster administrator of OpenShift, I expect user-defined tags added in kube resources must not be overridden by update action or by removing user-defined tags from `.spec.platformSpec.aws.resourceTags`.
+
+### API Extensions
+
+This proposal edits `Infrastructure.config.openshift.io/v1` type
+
+The Infrastructure resource example to use spec for api changes
+
+```go
+// AWSPlatformSpec holds the desired state of the Amazon Web Services Infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+    // Existing fields
+    ...
+    // ResourceTags is a list of additional tags to apply to AWS resources created for the cluster.
+    // See https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html for information on tagging AWS resources.
+    // AWS supports a maximum of 10 tags per resource. OpenShift reserves 5 tags for its use, leaving 5 tags
+    // available for the user.
+    // ResourceTags field is mutable and items can be removed.
+    // When a tag is removed from ResourceTags, the tag still persists on AWS resources.
+    // +kubebuilder:validation:MaxItems=10
+    // +optional
+    ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
+}
+
+// Existing type
+// AWSResourceTag is a tag to apply to AWS resources created for the cluster.
+type AWSResourceTag struct {
+    // key is the key of the tag
+    // +kubebuilder:validation:Required
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=128
+    // +kubebuilder:validation:Pattern=`^(?!openshift.io)(?!kubernetes.io)([0-9A-Za-z_.:\/=+-@]+$)`
+    // +required
+    Key string `json:"key"`
+    // value is the value of the tag.
+    // Some AWS service do not support empty values. Since tags are added to resources in many services, the
+    // length of the tag value must meet the requirements of all services.
```

Comment:
As above, include the requirements for the characters that are allowed and the length please. Remember to write it in a user friendly way